main.cpp

```diff
@@ -56,48 +56,77 @@ Purpose:  This project will show you the difference between member functions and
 
 #include <iostream>
 #include <string>
+
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    T(int v, const char* w) : value(v), name(w) {}
+
+    int value;
+    std::string name;
 };
 
-struct <#structName1#>                                //4
+struct Comparison
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare(T* a, T* b)
     {
-        if( a->value < b->value ) return a;
-        if( a->value > b->value ) return b;
+        if( a!=nullptr && b!=nullptr )
+        {
+            if( a->value < b->value ) return a;
+            if( a->value > b->value ) return b;
+        }
         return nullptr;
     }
 };
 
 struct U
 {
-    float <#name1#> { 0 }, <#name2#> { 0 };
-    <#returnType#> <#memberFunction#>(<#type name#>* <#updatedValue#>)      //12
+    float a { 0 }, b { 0 };
+    float makeCloser(float* newValue)
     {
-        
+        std::cout << "U's a value: " << a << std::endl;
+        if (newValue != nullptr)
+        {
+            a = *newValue;
+            std::cout << "U's updated a value: " << a << std::endl;
+            while( std::abs(b - a) > 0.001f )
+            {
+                if (b < a) 
+                {
+                    b += 1.0f;
+                }
+                else
+                {
+                    b -= 0.01f;
+                }
+            }
+            std::cout << "U's b updated value: " << b << std::endl;
+            return a*b;
+        }
+        else
+        return 0.f;
```

Comment:
There is no need to write `else` because the `if` block ends with `return`.
Also, if you did need to write `else`, the block for it should be indented.
- [ ] revise: remove `else`